Hibernate Reactive - trouble using identity generation strategy

### Describe the bug

Our app has 2 tables - one is using `ReactiveIdentifierGenerator`, the other is using `IDENTITY`, with Hibernate Reactive 2.0.0.Alpha2 I get casting exception when creating new instance.

### Expected behavior

I realize there is https://github.com/quarkusio/quarkus/issues/31974 so that documentation can't be perfect yet, but I'd appreciate hit if the problem is on my side or Quarks side.

### Actual behavior

Exception message
```
15:16:07,950 INFO  [app] org.hibernate.HibernateException: java.lang.ClassCastException: class org.hibernate.id.insert.GetGeneratedKeysDelegate cannot be cast to class org.hibernate.reactive.id.insert.ReactiveInsertGeneratedIdentifierDelegate (org.hibernate.id.insert.GetGeneratedKeysDelegate and org.hibernate.reactive.id.insert.ReactiveInsertGeneratedIdentifierDelegate are in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @799f7e29)
15:16:07,950 INFO  [app] 	at org.hibernate.reactive.session.impl.ReactiveExceptionConverter.convert(ReactiveExceptionConverter.java:28)
15:16:07,950 INFO  [app] 	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.lambda$doFlush$29(ReactiveSessionImpl.java:1110)
15:16:07,951 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
15:16:07,951 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:950)
15:16:07,951 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2340)
15:16:07,951 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:144)
15:16:07,951 INFO  [app] 	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.doFlush(ReactiveSessionImpl.java:1105)
15:16:07,952 INFO  [app] 	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.reactiveFlush(ReactiveSessionImpl.java:1052)
15:16:07,952 INFO  [app] 	at io.smallrye.context.impl.wrappers.SlowContextualSupplier.get(SlowContextualSupplier.java:21)
15:16:07,952 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage.subscribe(UniCreateFromCompletionStage.java:24)
15:16:07,952 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,952 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.lambda$subscribe$0(UniRunSubscribeOn.java:27)
15:16:07,952 INFO  [app] 	at org.hibernate.reactive.context.impl.VertxContext.execute(VertxContext.java:90)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.subscribe(UniRunSubscribeOn.java:25)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
15:16:07,953 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,954 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
15:16:07,954 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
15:16:07,954 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:60)
15:16:07,954 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onItem(UniOperatorProcessor.java:47)
15:16:07,954 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:63)
15:16:07,954 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
15:16:07,954 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
15:16:07,955 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2325)
15:16:07,955 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:144)
15:16:07,955 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forward(UniCreateFromCompletionStage.java:51)
15:16:07,955 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage.subscribe(UniCreateFromCompletionStage.java:35)
15:16:07,955 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,955 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.lambda$subscribe$0(UniRunSubscribeOn.java:27)
15:16:07,955 INFO  [app] 	at org.hibernate.reactive.context.impl.VertxContext.execute(VertxContext.java:90)
15:16:07,955 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.subscribe(UniRunSubscribeOn.java:25)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,956 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:31)
15:16:07,957 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,958 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnCancellationCall.subscribe(UniOnCancellationCall.java:27)
15:16:07,958 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,958 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
15:16:07,958 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,958 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
15:16:07,958 INFO  [app] 	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
15:16:07,958 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:81)
15:16:07,959 INFO  [app] 	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
15:16:07,959 INFO  [app] 	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromCompletionStage$CompletionStageUniSubscription.forwardResult(UniCreateFromCompletionStage.java:63)
15:16:07,959 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
15:16:07,959 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
15:16:07,959 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
15:16:07,959 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
15:16:07,960 INFO  [app] 	at io.vertx.core.Future.lambda$toCompletionStage$3(Future.java:384)
15:16:07,960 INFO  [app] 	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
15:16:07,960 INFO  [app] 	at io.vertx.core.impl.future.FutureImpl$ListenerArray.onSuccess(FutureImpl.java:262)
15:16:07,960 INFO  [app] 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
15:16:07,960 INFO  [app] 	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
15:16:07,960 INFO  [app] 	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
15:16:07,960 INFO  [app] 	at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)
15:16:07,961 INFO  [app] 	at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)
15:16:07,961 INFO  [app] 	at io.vertx.sqlclient.impl.TransactionImpl.lambda$wrap$0(TransactionImpl.java:72)
15:16:07,961 INFO  [app] 	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
15:16:07,961 INFO  [app] 	at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
15:16:07,961 INFO  [app] 	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:86)
15:16:07,961 INFO  [app] 	at io.vertx.core.impl.DuplicatedContext.execute(DuplicatedContext.java:163)
15:16:07,961 INFO  [app] 	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:51)
15:16:07,962 INFO  [app] 	at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
15:16:07,962 INFO  [app] 	at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
15:16:07,962 INFO  [app] 	at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)
15:16:07,962 INFO  [app] 	at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:41)
15:16:07,962 INFO  [app] 	at io.vertx.core.impl.future.PromiseImpl.handle(PromiseImpl.java:23)
15:16:07,962 INFO  [app] 	at io.vertx.pgclient.impl.PgSocketConnection.lambda$doSchedule$2(PgSocketConnection.java:155)
15:16:07,962 INFO  [app] 	at io.vertx.sqlclient.impl.command.CommandResponse.fire(CommandResponse.java:46)
15:16:07,962 INFO  [app] 	at io.vertx.sqlclient.impl.SocketConnectionBase.handleMessage(SocketConnectionBase.java:292)
15:16:07,962 INFO  [app] 	at io.vertx.pgclient.impl.PgSocketConnection.handleMessage(PgSocketConnection.java:97)
15:16:07,963 INFO  [app] 	at io.vertx.sqlclient.impl.SocketConnectionBase.lambda$init$0(SocketConnectionBase.java:105)
15:16:07,963 INFO  [app] 	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
15:16:07,963 INFO  [app] 	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
15:16:07,963 INFO  [app] 	at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:390)
15:16:07,963 INFO  [app] 	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:157)
15:16:07,963 INFO  [app] 	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
15:16:07,963 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
15:16:07,963 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
15:16:07,964 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
15:16:07,964 INFO  [app] 	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
15:16:07,964 INFO  [app] 	at io.vertx.pgclient.impl.codec.PgEncoder.lambda$write$0(PgEncoder.java:98)
15:16:07,964 INFO  [app] 	at io.vertx.pgclient.impl.codec.PgCommandCodec.handleReadyForQuery(PgCommandCodec.java:139)
15:16:07,964 INFO  [app] 	at io.vertx.pgclient.impl.codec.PgDecoder.decodeReadyForQuery(PgDecoder.java:237)
15:16:07,964 INFO  [app] 	at io.vertx.pgclient.impl.codec.PgDecoder.channelRead(PgDecoder.java:96)
15:16:07,964 INFO  [app] 	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
15:16:07,964 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
15:16:07,964 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
15:16:07,964 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
15:16:07,964 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
15:16:07,964 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
15:16:07,965 INFO  [app] 	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
15:16:07,965 INFO  [app] 	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
15:16:07,965 INFO  [app] 	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
15:16:07,965 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
15:16:07,965 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
15:16:07,965 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
15:16:07,965 INFO  [app] 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
15:16:07,965 INFO  [app] 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
15:16:07,966 INFO  [app] 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
15:16:07,966 INFO  [app] 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
15:16:07,966 INFO  [app] 	at java.base/java.lang.Thread.run(Thread.java:833)
15:16:07,966 INFO  [app] Caused by: java.lang.ClassCastException: class org.hibernate.id.insert.GetGeneratedKeysDelegate cannot be cast to class org.hibernate.reactive.id.insert.ReactiveInsertGeneratedIdentifierDelegate (org.hibernate.id.insert.GetGeneratedKeysDelegate and org.hibernate.reactive.id.insert.ReactiveInsertGeneratedIdentifierDelegate are in unnamed module of loader io.quarkus.bootstrap.runner.RunnerClassLoader @799f7e29)
15:16:07,966 INFO  [app] 	at org.hibernate.reactive.engine.jdbc.mutation.internal.ReactiveMutationExecutorPostInsertSingleTable.executeReactive(ReactiveMutationExecutorPostInsertSingleTable.java:66)
15:16:07,966 INFO  [app] 	at org.hibernate.reactive.persister.entity.mutation.ReactiveInsertCoordinator.lambda$doStaticInserts$9(ReactiveInsertCoordinator.java:166)
15:16:07,966 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
15:16:07,967 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
15:16:07,967 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
15:16:07,967 INFO  [app] 	at org.hibernate.reactive.persister.entity.mutation.ReactiveInsertCoordinator.doStaticInserts(ReactiveInsertCoordinator.java:166)
15:16:07,967 INFO  [app] 	at org.hibernate.reactive.persister.entity.mutation.ReactiveInsertCoordinator.lambda$coordinateReactiveInsert$0(ReactiveInsertCoordinator.java:54)
15:16:07,967 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
15:16:07,967 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
15:16:07,967 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
15:16:07,967 INFO  [app] 	at org.hibernate.reactive.persister.entity.mutation.ReactiveInsertCoordinator.coordinateReactiveInsert(ReactiveInsertCoordinator.java:51)
15:16:07,967 INFO  [app] 	at org.hibernate.reactive.persister.entity.impl.ReactiveSingleTableEntityPersister.insertReactive(ReactiveSingleTableEntityPersister.java:242)
15:16:07,968 INFO  [app] 	at org.hibernate.reactive.engine.impl.ReactiveEntityIdentityInsertAction.lambda$reactiveExecute$0(ReactiveEntityIdentityInsertAction.java:65)
15:16:07,968 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
15:16:07,968 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
15:16:07,968 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
15:16:07,968 INFO  [app] 	at org.hibernate.reactive.engine.impl.ReactiveEntityIdentityInsertAction.reactiveExecute(ReactiveEntityIdentityInsertAction.java:65)
15:16:07,968 INFO  [app] 	at org.hibernate.reactive.engine.ReactiveActionQueue.lambda$executeActions$9(ReactiveActionQueue.java:646)
15:16:07,968 INFO  [app] 	at org.hibernate.reactive.util.impl.CompletionStages.lambda$loop$7(CompletionStages.java:373)
15:16:07,968 INFO  [app] 	at org.hibernate.reactive.util.impl.CompletionStages$ArrayLoop.next(CompletionStages.java:447)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.util.async.impl.AsyncTrampoline.lambda$asyncWhile$1(AsyncTrampoline.java:215)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.unroll(AsyncTrampoline.java:121)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.util.async.impl.AsyncTrampoline$TrampolineInternal.trampoline(AsyncTrampoline.java:102)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.util.async.impl.AsyncTrampoline.asyncWhile(AsyncTrampoline.java:197)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.util.async.impl.AsyncTrampoline.asyncWhile(AsyncTrampoline.java:215)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.util.impl.CompletionStages.loop(CompletionStages.java:374)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.util.impl.CompletionStages.loop(CompletionStages.java:345)
15:16:07,969 INFO  [app] 	at org.hibernate.reactive.engine.ReactiveActionQueue.executeActions(ReactiveActionQueue.java:643)
15:16:07,970 INFO  [app] 	at org.hibernate.reactive.engine.ReactiveActionQueue.lambda$executeActions$7(ReactiveActionQueue.java:543)
15:16:07,970 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
15:16:07,970 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
15:16:07,970 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
15:16:07,970 INFO  [app] 	at org.hibernate.reactive.engine.ReactiveActionQueue.executeActions(ReactiveActionQueue.java:543)
15:16:07,970 INFO  [app] 	at org.hibernate.reactive.event.impl.AbstractReactiveFlushingEventListener.lambda$performExecutions$0(AbstractReactiveFlushingEventListener.java:64)
15:16:07,970 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
15:16:07,970 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
15:16:07,970 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
15:16:07,971 INFO  [app] 	at org.hibernate.reactive.event.impl.AbstractReactiveFlushingEventListener.performExecutions(AbstractReactiveFlushingEventListener.java:58)
15:16:07,971 INFO  [app] 	at org.hibernate.reactive.event.impl.DefaultReactiveFlushEventListener.lambda$reactiveOnFlush$0(DefaultReactiveFlushEventListener.java:40)
15:16:07,971 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
15:16:07,971 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
15:16:07,971 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
15:16:07,971 INFO  [app] 	at org.hibernate.reactive.event.impl.DefaultReactiveFlushEventListener.reactiveOnFlush(DefaultReactiveFlushEventListener.java:40)
15:16:07,972 INFO  [app] 	at org.hibernate.event.service.internal.EventListenerGroupImpl.lambda$fireEventOnEachListener$0(EventListenerGroupImpl.java:153)
15:16:07,972 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
15:16:07,972 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2309)
15:16:07,972 INFO  [app] 	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:144)
15:16:07,972 INFO  [app] 	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:153)
15:16:07,972 INFO  [app] 	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.doFlush(ReactiveSessionImpl.java:1104)
15:16:07,972 INFO  [app] 	... 115 more
```

### How to Reproduce?

I've got 2 reproducers. First one is for the exception described above:
1. `git clone git@github.com:michalvavrik/quarkus-test-suite.git`
2. `cd quarkus-test-suite`
3. `git checkout enable-hibernate-reactive-modules`
4. `cd sql-db/hibernate-reactive`
5. `mvn clean verify -Dit.test=PostgresqlDatabaseHibernateReactiveIT#createBookWithGeneratedId`

I also tried to compare my test with Quarkus integration tests and when I added `@GeneratedValue(strategy = GenerationType.IDENTITY)` as you can see here https://github.com/quarkusio/quarkus/compare/main...michalvavrik:quarkus:reproducer/hibernate-reactive-identity, I got exception `org.hibernate.HibernateException: java.util.concurrent.CompletionException: java.lang.ClassCastException: class org.hibernate.id.IdentityGenerator cannot be cast to class org.hibernate.id.IdentifierGenerator`. I don't understand why shouldn't I be able to use identity as I can see that Hibernate Reactive own tests are using it without extra configuration, f.e. https://github.com/hibernate/hibernate-reactive/blob/main/hibernate-reactive-core/src/test/java/org/hibernate/reactive/ReactiveStatelessProxyUpdateTest.java#L180

### Output of `uname -a` or `ver`

Linux

### Output of `java -version`

openjdk 17.0.4 2022-07-19

### GraalVM version (if different from Java)

OpenJDK Runtime Environment GraalVM CE 23.2

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

_No response_

--- a/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/QuarkusStaticInitDialectFactory.java
+++ b/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/service/QuarkusStaticInitDialectFactory.java
@@ -14,7 +14,7 @@ import org.hibernate.service.spi.ServiceRegistryAwareService;
  * the same as Hibernate ORM's default one except it records the dialect
  * so that we can reuse it at runtime init.
  */
-public final class QuarkusStaticInitDialectFactory extends DialectFactoryImpl
+public class QuarkusStaticInitDialectFactory extends DialectFactoryImpl
         implements DialectFactory, ServiceRegistryAwareService {
 
     private Dialect dialect;

--- a/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/boot/registry/ReactiveHibernateInitiatorListProvider.java
+++ b/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/boot/registry/ReactiveHibernateInitiatorListProvider.java
@@ -34,9 +34,9 @@ import io.quarkus.hibernate.orm.runtime.service.InitialInitiatorListProvider;
 import io.quarkus.hibernate.orm.runtime.service.QuarkusImportSqlCommandExtractorInitiator;
 import io.quarkus.hibernate.orm.runtime.service.QuarkusMutableIdentifierGeneratorFactoryInitiator;
 import io.quarkus.hibernate.orm.runtime.service.QuarkusRegionFactoryInitiator;
-import io.quarkus.hibernate.orm.runtime.service.QuarkusStaticInitDialectFactoryInitiator;
 import io.quarkus.hibernate.orm.runtime.service.StandardHibernateORMInitiatorListProvider;
 import io.quarkus.hibernate.reactive.runtime.customized.QuarkusNoJdbcConnectionProviderInitiator;
+import io.quarkus.hibernate.reactive.runtime.customized.QuarkusReactiveDialectFactoryInitiator;
 
 /**
  * Defines the initial list of StandardServiceInitiator instances used to initialize the
@@ -86,8 +86,8 @@ public final class ReactiveHibernateInitiatorListProvider implements InitialInit
         serviceInitiators.add(MultiTenantConnectionProviderInitiator.INSTANCE);
         serviceInitiators.add(DialectResolverInitiator.INSTANCE);
 
-        // Custom Quarkus implementation !
-        serviceInitiators.add(QuarkusStaticInitDialectFactoryInitiator.INSTANCE);
+        // Custom Quarkus V Reactive implementation !
+        serviceInitiators.add(QuarkusReactiveDialectFactoryInitiator.INSTANCE);
 
         // Default implementation
         serviceInitiators.add(BatchBuilderInitiator.INSTANCE);

--- a/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/customized/QuarkusReactiveDialectFactoryInitiator.java
+++ b/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/customized/QuarkusReactiveDialectFactoryInitiator.java
@@ -1,0 +1,22 @@
+package io.quarkus.hibernate.reactive.runtime.customized;
+
+import java.util.Map;
+
+import org.hibernate.boot.registry.StandardServiceInitiator;
+import org.hibernate.engine.jdbc.dialect.spi.DialectFactory;
+import org.hibernate.service.spi.ServiceRegistryImplementor;
+
+public final class QuarkusReactiveDialectFactoryInitiator implements StandardServiceInitiator<DialectFactory> {
+
+    public static final QuarkusReactiveDialectFactoryInitiator INSTANCE = new QuarkusReactiveDialectFactoryInitiator();
+
+    @Override
+    public Class<DialectFactory> getServiceInitiated() {
+        return DialectFactory.class;
+    }
+
+    @Override
+    public DialectFactory initiateService(Map configurationValues, ServiceRegistryImplementor registry) {
+        return new ReactiveQuarkusStaticInitDialectFactory();
+    }
+}

--- a/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/customized/ReactiveQuarkusStaticInitDialectFactory.java
+++ b/extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/customized/ReactiveQuarkusStaticInitDialectFactory.java
@@ -1,0 +1,30 @@
+package io.quarkus.hibernate.reactive.runtime.customized;
+
+import java.util.Map;
+
+import org.hibernate.HibernateException;
+import org.hibernate.dialect.Dialect;
+import org.hibernate.engine.jdbc.dialect.spi.DialectFactory;
+import org.hibernate.engine.jdbc.dialect.spi.DialectResolutionInfoSource;
+import org.hibernate.reactive.dialect.ReactiveDialectWrapper;
+import org.hibernate.service.spi.ServiceRegistryAwareService;
+
+import io.quarkus.hibernate.orm.runtime.service.QuarkusStaticInitDialectFactory;
+
+public class ReactiveQuarkusStaticInitDialectFactory extends QuarkusStaticInitDialectFactory
+        implements DialectFactory, ServiceRegistryAwareService {
+
+    private Dialect wrapDialect;
+
+    @Override
+    public Dialect buildDialect(Map<String, Object> map, DialectResolutionInfoSource dialectResolutionInfoSource)
+            throws HibernateException {
+        this.wrapDialect = new ReactiveDialectWrapper(super.buildDialect(map, dialectResolutionInfoSource));
+        return this.wrapDialect;
+    }
+
+    @Override
+    public Dialect getDialect() {
+        return this.wrapDialect;
+    }
+}

